File: summer23InstaCult.client/src/models/Cult.js
import { Profile } from "./Account.js"
import { RepoItem } from "./RepoItem.js"

export class Cult extends RepoItem {
  constructor (data) {
    super(data)
    // this.id = data.id
    // this.createdAt = new Date(data.createdAt)
    // this.updatedAt = new Date(data.updatedAt)
    this.name = data.name
    this.cultistCount = data.cultistCount
    this.description = data.description
    this.fee = data.fee
    this.coverImg = data.coverImg
    this.leaderId = data.leaderId
    this.leader = new Profile(data.leader)
  }
}

File: summer23InstaCult/Models/Cult.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace summer23InstaCult.Models;

public class Cult : RepoItem<int>
{
  // public int Id { get; set; }
  public string Name { get; set; }
  public string Description { get; set; }
  public int Fee { get; set; }
  public string CoverImg { get; set; }
  public string LeaderId { get; set; }
  public Profile Leader { get; set; }
  public int CultistCount { get; set; }
  // public DateTime CreatedAt { get; set; }
  // public DateTime UpdatedAt { get; set; }
}
